### DISCUSSION TITLE
How to use CrazyS for Control Barrier Function simulation?

### DISCUSSION THREAD

**Discussion by hoangah2000vn** [gsilano/CrazyS#99] [General]
Hi! I'm currently working on a multiagent control project in which the aim is to have multiple crazyflie flying without collisions using the Control Barrier Function (CBF). The CBF is trained using the neural network so after training it is stored as neural network parameters. My questions is how do I integrate the CBF in Crazyflie hardware. Should I modify the controller code?

Please let me know if any part of my question is not clear!

**Comment by gsilano** [comment]
Hi! I don't know if you can integrate it into the Crazyflie microcontroller or not. It looks like a technical question for [Bitcraze](https://github.com/bitcraze).

If you want to test it with CrazyS, I could give you advice on how to proceed.

**Reply by hoangah2000vn** [reply]
Hi yes could you give me advice on how to proceed? what should I modify?

**Reply by gsilano** [reply]
Ok! Have you already read the related papers?

**Reply by hoangah2000vn** [reply]
yes I did

**Reply by gsilano** [reply]
Sorry for the late reply, but this period is a bit busy for me. To answer your question I would need to know what your NN outputs are. In other words, the I/O interface. As you have probably read from the reference publications, in CrazyS you can control the behavior of the vehicle by providing the propeller angular speeds.

**Comment by hoangah2000vn** [comment]
Hi bumping this
